pyxem/signals/segments.py

```diff
@@ -0,0 +1,443 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+"""Signal classes for segmentation results obtained from
+i) machine learning (LearningSegment) and ii) VDF images (VDFSegment).
+
+"""
+
+import numpy as np
+from tqdm import tqdm
+
+from hyperspy.signals import Signal2D
+
+from pyxem.utils.segment_utils import (norm_cross_corr, separate_watershed,
+                                       get_gaussian2d)
+from pyxem.signals.diffraction_vectors import DiffractionVectors
+from pyxem.signals.electron_diffraction2d import ElectronDiffraction2D
+from pyxem.signals import transfer_signal_axes
+
+from traits.trait_errors import TraitError
+
+
+class LearningSegment:
```

Comment:
Presumably .decompostion() methods should now be returning in this class?

Reply:
Indeed perhaps this is basically containing the same information as decomposition results - are there classes specifically for that in hyperspy and if so this should be inheriting from that probably.

Reply:
Ok, I think we should consider whether it makes more sense for this to be a SEDLearningResults class that inherits from LearningResults and adds these specific things that make sense for that data. But possibly that can wait for the future.